nebula-logger/main/logger-engine/classes/ComponentLogger.cls

```diff
@@ -0,0 +1,269 @@
+            throw new AuraHandledException(apexException.getMessage());
+        }
+    }
+
+    private static void setComponentErrorDetails(LogEntryEventBuilder logEntryEventBuilder, ComponentError componentError) {
+        if (componentError == null) {
+            return;
+        }
+
+        logEntryEventBuilder.getLogEntryEvent().ExceptionMessage__c = componentError.message;
+        logEntryEventBuilder.getLogEntryEvent().ExceptionStackTrace__c = componentError.stack;
+        logEntryEventBuilder.getLogEntryEvent().ExceptionType__c = EXCEPTION_TYPE_PREFIX + componentError.type;
+    }
+
+    private static void setStackTraceDetails(LogEntryEventBuilder logEntryEventBuilder, String stackTraceString) {
+        String originLocation;
+        String originType;
+        Boolean isAuraComponent = false;
+
+        if (String.isNotBlank(stackTraceString) == true) {
+            String loggerComponentReferenceToIgnore = '/c/logger.js';
+            String auraComponentContent = '/n/components/';
+            String lwcModuleContent = '/n/modules/';
+            Boolean parentComponentFound = false;
+
+            List<String> stackTraceLines = new List<String>();
+            for (String currentStackTraceLine : stackTraceString.split('\n')) {
+                if (currentStackTraceLine.contains(loggerComponentReferenceToIgnore) == true) {
+                    continue;
+                } else if (parentComponentFound == false && currentStackTraceLine.contains(auraComponentContent) == true) {
+                    isAuraComponent = true;
+                    parentComponentFound = true;
+                } else if (parentComponentFound == false && currentStackTraceLine.contains(lwcModuleContent) == true) {
+                    parentComponentFound = true;
+                }
+
+                if (parentComponentFound == true) {
+                    stackTraceLines.add(currentStackTraceLine);
+                }
+            }
+
+            stackTraceString = String.join(stackTraceLines, '\n');
+            if (String.isNotBlank(stackTraceString)) {
+                String componentName;
+                String componentFunction;
+                if (isAuraComponent == true) {
+                    componentName = stackTraceLines.get(0).substringAfterLast(auraComponentContent).substringBefore('.js');
+                    componentFunction = stackTraceLines.get(0).substringBefore(' (').substringAfter('at ');
+                } else {
+                    componentName = stackTraceLines.get(0).substringAfterLast(lwcModuleContent).substringBefore('.js');
+                    componentFunction = stackTraceLines.get(0).substringBefore(' (').substringAfter('at ').substringAfter('.');
+                }
+
+                originLocation = componentName + '.' + componentFunction;
+            }
+        }
+
+        logEntryEventBuilder.getLogEntryEvent().ComponentType__c = isAuraComponent == true ? 'Aura' : 'Web';
```

Comment:
since `LWC` is so synonymous with web components, I'm wondering whether or not you're open to changing `Web` to that

Reply:
I was kind of hoping you might mention this! I couldn't decide if I liked `Web` or `LWC` better, but I think you're right - `LWC` is more intuitive/explicit/synonymous, so I'll change the picklist value to `LWC` instead.